test/e2e/nodelatencymonitor_test.go

```diff
@@ -0,0 +1,131 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	clientset "antrea.io/antrea/pkg/client/clientset/versioned"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetKubeconfigPath returns the path to the kubeconfig file used for connecting to the Kubernetes cluster.
+func GetKubeconfigPath() (string, error) {
+	// Check if the KUBECONFIG environment variable is set
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		// Default to the kubeconfig path in the user's home directory if not set
+		kubeconfig = filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	}
+
+	// Verify that the kubeconfig file exists
+	if _, err := os.Stat(kubeconfig); os.IsNotExist(err) {
+		return "", fmt.Errorf("kubeconfig file does not exist at path: %s", kubeconfig)
+	}
+
+	return kubeconfig, nil
+}
+
+// getAntreaClient returns a clientset for interacting with Antrea CRDs
+func getAntreaClient(t *testing.T) *clientset.Clientset {
+	kubeconfig, err := GetKubeconfigPath()
+	failOnError(err, t)
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	failOnError(err, t)
+
+	antreaClientset, err := clientset.NewForConfig(config)
+	failOnError(err, t)
+	return antreaClientset
+}
```

Comment:
Please refer to other e2e tests. The framework already provides access to clientsets for K8s / Antrea, and there is no need for these 2 functions.

Reply:
Hii @antoninbas  i'm so close towards my goal but 
getting this error in test file 
nodelatencymonitor_test.go:143: Failed to retrieve NodeLatencyStats: the server could not find the requested resource (get nodelatencystats.crd.antrea.io)
also 
 kubectl get nodelatencystats
error: the server doesn't have a resource type "nodelatencystats"

is there any thing i have missed?

Reply:
`nodelatencystats.crd.antrea.io` does not exist as `NodeLatencyStats` is not a CRD

`kubectl get nodelatencystats` should work fine unless you are using an older version of Antrea (prior to v2.1). You should be testing against the "latest" Antrea (main branch).

When running `kubectl api-resources`, you should see the following:
```
...
antreaclusternetworkpolicystats                stats.antrea.io/v1alpha1          false        AntreaClusterNetworkPolicyStats
antreanetworkpolicystats                       stats.antrea.io/v1alpha1          true         AntreaNetworkPolicyStats
multicastgroups                                stats.antrea.io/v1alpha1          false        MulticastGroup
networkpolicystats                             stats.antrea.io/v1alpha1          true         NetworkPolicyStats
nodelatencystats                               stats.antrea.io/v1alpha1          false        NodeLatencyStats
...
```